plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AmazonS3Fixture.java

```diff
@@ -322,12 +418,48 @@ public static void main(final String[] args) throws Exception {
         /** Bucket name **/
         final String name;
 
+        final String key;
+
+        final String token;
+
         /** Blobs contained in the bucket **/
         final Map<String, byte[]> objects;
 
-        Bucket(final String name) {
-            this.name = Objects.requireNonNull(name);
+        Bucket(final Properties properties, Map<String, Bucket> buckets, final String prefix) {
+            this(properties, buckets, prefix, true);
+        }
+
+        Bucket(final Properties properties,
+               final Map<String, Bucket> buckets,
+               final String prefix,
+               final boolean tokenRequired) {
+            this(properties, buckets, prefix, tokenRequired, null);
+        }
+
+        Bucket(final Properties properties,
+               final Map<String, Bucket> buckets,
+               final String prefix,
+               final Random random) {
+            this(properties, buckets, prefix, false, random);
+        }
+
+        private Bucket(final Properties properties,
```

Comment:
I think it'd be clearer to simply inline these overloads and spell out all the parameters at the call sites. We only make three of these objects, so four overloads of the constructor seems excessive :)

Also if `Bucket` were not `static` then you wouldn't need to pass `buckets` in.

Reply:
makes sence